.github/workflows/on-pr.yml

```
@@ -122,7 +122,7 @@ jobs:
     needs:
       - should-run
       - build-test
-    if: ${{ needs.should-run.outputs.go == 'true' && contains(fromJSON('["release", "master"]'), github.ref_name) }}
+    if: ${{ needs.should-run.outputs.go == 'true' && (startsWith(github.base_ref, 'release') || startsWith(github.base_ref, 'master')) }}
```

```suggestion
    if: ${{ needs.should-run.outputs.go == 'true' && (startsWith(github.base_ref, 'release') || github.base_ref == 'master') }}
```

this will match `on-trigger` better, and makes a bit more sense

---

I like it, thanks.

---

BTW, we won't be merging back into the `master` branch going forward, so it might make sense to just remove it.

---

Well, I think it should be a separate decision and should be kept in sync in both files, so let's merge this as is right now, wdyt?

---

SGTM.